core/string/encode_spec.rb

```diff
@@ -60,6 +60,12 @@
 
       "\rfoo".encode(universal_newline: true).should == "\nfoo"
     end
+
+    it "replaces invalid encoding" do
+      invalid = "\xe3\x81\xa1\xE3\x81\xFF".force_encoding("UTF-8")
+      encoded = invalid.encode("UTF-16LE", invalid: :replace, replace: '?').encode("UTF-8")
```

Comment:
This is doing 2 `encode` calls on the same line, could you check the result of the first encode call as well?
That would help when fixing the implementation to pass this spec.